core/inspector.rs

```diff
@@ -610,24 +656,34 @@ impl task::ArcWake for InspectorWaker {
   }
 }
 
+#[derive(Debug)]
+pub enum InspectorSessionKind {
+  Blocking,
+  NonBlocking { wait_for_disconnect: bool },
+}
+
 /// An inspector session that proxies messages to concrete "transport layer",
 /// eg. Websocket or another set of channels.
 struct InspectorSession {
+  is_dispatching_message: Rc<AtomicBool>,
```

Comment:
Do we really need an `AtomicBool` here? Why can't it keep using `RefCell<bool>`? Unless `AtomicBool` is faster then I don't care